From 0ddf866b4a187b8b44fa147add13c2cf9a1a28c3 Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Thu, 10 Aug 2017 12:53:52 +0530
Subject: [PATCH 285/532] v4l: xilinx: csi2rxss: Handle failing to create
 custom control

commit 9921a9274ababfa83d9f38ead1d38b77e592f5b7 from
https://github.com/Xilinx/linux-xlnx.git

Correctly handle the error generated while creating new custom control.

Signed-off-by: Vishal Sagar <vsagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-csi2rxss.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/media/platform/xilinx/xilinx-csi2rxss.c b/drivers/media/platform/xilinx/xilinx-csi2rxss.c
index a102741..28724f2 100644
--- a/drivers/media/platform/xilinx/xilinx-csi2rxss.c
+++ b/drivers/media/platform/xilinx/xilinx-csi2rxss.c
@@ -1675,6 +1675,11 @@ static int xcsi2rxss_probe(struct platform_device *pdev)
 				i, xcsi2rxss_ctrls[i].id);
 		ctrl = v4l2_ctrl_new_custom(&xcsi2rxss->ctrl_handler,
 						&xcsi2rxss_ctrls[i], NULL);
+		if (!ctrl) {
+			dev_err(xcsi2rxss->core.dev, "Failed for %s ctrl\n",
+				xcsi2rxss_ctrls[i].name);
+			goto error;
+		}
 	}
 
 	dev_dbg(xcsi2rxss->core.dev, "# v4l2 ctrls registered = %d\n", i - 1);
-- 
2.7.4

